edge-util/src/Microsoft.Azure.Devices.Edge.Util/ClientTokenProvider.cs

```diff
@@ -60,5 +87,34 @@ public async Task<string> GetTokenAsync(Option<TimeSpan> ttl)
                 throw new TokenProviderException(e);
             }
         }
+
+        class TokenCache
+        {
+            readonly TimeSpan expiryBuffer;
+
+            Option<string> token;
+            DateTime tokenExpiry;
+
+            public TokenCache(TimeSpan expiryBuffer)
+            {
+                this.expiryBuffer = expiryBuffer;
+            }
+
+            public void UpdateToken(string token, DateTime tokenExpiry)
+            {
+                this.token = Option.Maybe(token);
+                this.tokenExpiry = tokenExpiry;
+            }
+
+            public Option<string> GetToken()
+            {
+                if (this.tokenExpiry - DateTime.UtcNow > this.expiryBuffer)
```

Comment:
it's it possible for DateTime.Now to go over tokenExpiry? it seems that would be possible if GetToken doesn't get called for a while so it's not triggering update.